src/transport/SecureSessionMgr.cpp

```diff
@@ -205,6 +205,16 @@ void SecureSessionMgr::HandleDataReceived(const MessageHeader & header, const Pe
     {
         if (!connection->mPeerConnections.FindPeerConnectionState(peerAddress, &state))
         {
+            if (header.GetSourceNodeId().HasValue())
+            {
+                // If the data is from a new address BUT the node id is the same as a previous
+                // connection, mark the previous connection invalid in order to not have duplicate node ids.
+                if (connection->mPeerConnections.FindPeerConnectionState(header.GetSourceNodeId().Value(), &state))
+                {
+                    connection->mPeerConnections.MarkConnectionInvalid(state);
```

Comment:
We may want a callback when this happens. For any persistent channels (TCP or BLE) we will need to close the connection.

Reply:
Any objections if I change the name of the current callbacks ?

 I was thinking of something like:
  OnMessageReceived -> OnMessage
  OnReceivedError -> OnError

It would let me reuse the OnError callback and the embedder can then decide what to do depending on the type of errors.

Reply:
Seems reasonable. Can we somehoe give the OnError callback context to detect when a 'overwrite/cancel' is provided?

Maybe OnExpiry then is also OnError.

Reply:
@vivien-apple ?

Reply:
@woody-apple Yeah sorry for the latency I'm trying to figure out what I want exactly. 

@andy31415 OnExpiry ? Does it exists in the tree or did you mean something else ?

Reply:
Connection expiry exists currently in the tree and is treated reasonably stand-alone:

https://github.com/project-chip/connectedhomeip/blob/master/src/transport/SecureSessionMgr.h#L182

we may need some thought since this particular type of error has sideeffects (need to free up resources on the connection)... but maybe any error on a connection should close/free it.

Reply:
I also see we are piling up changes and wondering if we should try to make some changes here or file an issue and work on these things separately.

Reply:
Right. So I end up just calling the expiry callback when this thing happens. Would be better to do it separately.